RationaleMCP/0027/HindleyMilner.md

```diff
@@ -0,0 +1,201 @@
+# Unit checking
+Unit checking is performed by using a type inference algorithm, albeit adapted to work with units. An example of such an algorithm is the Hindley-Milner algorithm.
```

Comment:
Maybe you could make a stronger statement about the connection to Hindley-Milner?  Like saying that the algorithm proposed here has been inspired by Hindley-Milner?